packages/mobile/src/components/ContactCircle.tsx

```diff
@@ -0,0 +1,112 @@
+import colors from '@celo/react-components/styles/colors'
+import fontStyles from '@celo/react-components/styles/fonts'
+import { getContactNameHash } from '@celo/utils/src/contacts'
+import * as React from 'react'
+import { Image, StyleSheet, Text, View, ViewStyle } from 'react-native'
+import { MinimalContact } from 'react-native-contacts'
+import { useSelector } from 'react-redux'
+import DefaultAvatar from 'src/icons/DefaultAvatar'
+import { addressToDisplayNameSelector } from 'src/identity/reducer'
+
+interface Props {
+  style?: ViewStyle
+  contact?: MinimalContact
+  name: string | null // Requiring a value so we need to be explicit if we dont have it
+  address?: string
+  size?: number
+  thumbnailPath?: string | null
+}
+
+const DEFAULT_ICON_SIZE = 40
+export const contactIconColors = [colors.teal, colors.orange, colors.purple]
+
+const getAddressBackgroundColor = (address: string) =>
+  `hsl(${parseInt(address.substring(0, 5), 16) % 360}, 53%, 93%)`
+const getAddressForegroundColor = (address: string) =>
+  `hsl(${parseInt(address.substring(0, 5), 16) % 360}, 67%, 24%)`
+const getContactColor = (contact: MinimalContact) =>
+  getAddressBackgroundColor(getContactNameHash(contact))
+const getContactInitial = (contact: MinimalContact) => getNameInitial(contact.displayName)
+const getNameInitial = (name: string) => name.charAt(0).toLocaleUpperCase()
+
+function ContactCircle({ contact, size, thumbnailPath, name, address, style }: Props) {
+  const addressToDisplayName = useSelector(addressToDisplayNameSelector)
+  const addressInfo = addressToDisplayName[address || '']
```

Comment:
nit: I think we can write it as `address && addressToDisplayName[address]`, it feel a bit more explicit what we are doing here. wdyt?